src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.ts

```diff
@@ -294,12 +294,19 @@ export function getCategoryOrderByGenericAssayType(genericAssayType: string) {
         : undefined;
 }
 
+export function constructGeneRegex(hugoGeneSymbol: string) {
+    // Match whole gene symbol text and case-insensitive, non-alphanumeric characters is allowed after gene symbol
+    // Sometimes, gene symbol might appear in description text
+    // If it shows in description text, we might want to find if it's before a blank space
+    return new RegExp(`^(${hugoGeneSymbol}|${hugoGeneSymbol}\\W+.*)$`, 'i');
```

Comment:
wouldn't it be enough to just detect the symbol at the beginning of text?  I mean, why do you care what happens after the symbol?

Reply:
@alisman It's not enough because we want to keep `TP53` but filter out `TP53BP1`, so I must see what's following the desired symbol.